### PR TITLE
Move RangeHelper to a shared sources package

### PR DESCRIPTION
**Pull request by jbagga**
Move byte range helpers to a shared source package so MVC can use it

cc @pranavkm

### CODE REVIEW

**Comment by pranavkm** [comment]
Missing this - https://github.com/aspnet/Routing/blob/dev/NuGetPackageVerifier.json#L2-L7

File: shared/Microsoft.AspNetCore.StaticFiles.ByteRangeHelper.Sources/ByteRangeHelper.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@
 using System.Collections.Generic;
 using Microsoft.Net.Http.Headers;
 
-namespace Microsoft.AspNetCore.StaticFiles.Infrastructure
+namespace Microsoft.AspNetCore.StaticFiles
```
Comment (pranavkm) [inline shared/Microsoft.AspNetCore.StaticFiles.ByteRangeHelper.Sources/ByteRangeHelper.cs]: You could change the ns to `Microsoft.Net.Http.Headers`. It doesn't have much to do with StaticFiles apart from it being located here.

Comment (pranavkm) [inline shared/Microsoft.AspNetCore.StaticFiles.ByteRangeHelper.Sources/ByteRangeHelper.cs]: Did you speak to @Tratcher about possibly moving this in to HttpAbstractions?

Comment (Tratcher) [inline shared/Microsoft.AspNetCore.StaticFiles.ByteRangeHelper.Sources/ByteRangeHelper.cs]: Yes please.

**Comment by jbagga** [comment]
Closing as this is now being addressed in https://github.com/aspnet/HttpAbstractions/pull/807

**Comment by jbagga** [comment]
See https://github.com/aspnet/StaticFiles/pull/185

**Pull request closed without merging**